web/src/main/java/org/mskcc/cbio/portal/web/ProxySessionServiceController.java

```diff
@@ -268,63 +289,177 @@ public void updateUsersInVirtualStudy(@PathVariable SessionType type, @PathVaria
 			@PathVariable String id,
             @RequestBody  JSONObject  body,
 			HttpServletResponse response) throws IOException {
 
-		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-
-		// updates only allowed for type group for now
-		List<SessionType> validSessionTypes = Arrays.asList(SessionType.group);
+		// updates only allowed for type group and settings
+		List<SessionType> validSessionTypes = Arrays.asList(SessionType.group, SessionType.settings);
 
-		if (validSessionTypes.contains(type) && isSessionServiceEnabled() && !(authentication == null
-				|| (authentication instanceof AnonymousAuthenticationToken))) {
+		if (validSessionTypes.contains(type) && isAuthorized()) {
 
 			ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES,
 					false);
-			String virtualStudyStr = mapper.writeValueAsString(getSession(type, id, response));
-			VirtualStudy virtualStudy = mapper.readValue(virtualStudyStr, VirtualStudy.class);
-
-			String user = ((UserDetails) authentication.getPrincipal()).getUsername();
-			//only allow owner to update virtual study
-			if (user.equals(virtualStudy.getData().getOwner()) && virtualStudy.getType().equals(type.toString())) {
-
-				VirtualStudyData updatedVirtualStudyData = mapper.readValue(body.toString(), VirtualStudyData.class);
-				updatedVirtualStudyData.setCreated(virtualStudy.getData().getCreated());
-				updatedVirtualStudyData.setOwner(virtualStudy.getData().getOwner());
-				updatedVirtualStudyData.setUsers(virtualStudy.getData().getUsers());
-
-				virtualStudy.setData(updatedVirtualStudyData);
-				RestTemplate restTemplate = new RestTemplate();
-				HttpEntity<Object> httpEntity = new HttpEntity<Object>(updatedVirtualStudyData, getHttpHeaders());
-				
-				restTemplate.put(sessionServiceURL + type + "/" + id, httpEntity);
-				response.setStatus(HttpStatus.OK.value());
-			} else {
-				response.setStatus(HttpStatus.UNAUTHORIZED.value());
-			}
+			
+			Session session = getSession(type, id, response).getBody();
+			
+			if(session.getType().equals(SessionType.group)) {
+			    VirtualStudy virtualStudy = mapper.readValue(mapper.writeValueAsString(session), VirtualStudy.class);
+			    
+			    VirtualStudyData virtualStudyData = virtualStudy.getData();;
+	            //only allow owner to update virtual study
+	            if (userName().equals(virtualStudyData.getOwner()) && virtualStudy.getType().equals(type)) {
+
+	                VirtualStudyData updatedVirtualStudyData = mapper.readValue(body.toString(), VirtualStudyData.class);
+	                updatedVirtualStudyData.setCreated(virtualStudyData.getCreated());
+	                updatedVirtualStudyData.setOwner(virtualStudyData.getOwner());
+	                updatedVirtualStudyData.setUsers(virtualStudyData.getUsers());
+
+	                
+	                virtualStudy.setData(mapper.writeValueAsString(updatedVirtualStudyData));
```

Comment:
Why do we need to update virtualStudy?

Reply:
this update is only for `group` which has the same format as `virtualStudy`